app/Http/Controllers/API/ChatController.php

```diff
@@ -105,10 +105,15 @@ public function privateMessages()
 
     public function postMessage()
     {
-        if (mb_strlen(Request::input('message'), 'UTF-8') >= 1024) {
+        $messagecontent = preg_replace('/[\x00-\x1f\x7F]/', '', Request::input('message'));
+
+        if (mb_strlen(Request::input('message'), 'UTF-8') >= 1000) {
             abort(422);
         }
 
+        if (Request::input('is_action') === 'True') {
+            $messagecontent = "\001ACTION ".$messagecontent."\001";
```

Comment:
make a mutator in model which accepts array of `content` and `is_action` which then clean up and set content accordingly. And `\x01` for consistency.

(put in trait for now until someone writes better abstraction between two Message models)